dev_guide/deployments/advanced_deployment_strategies.adoc

```diff
@@ -0,0 +1,295 @@
+----
+
+. In your browser, refresh the page until you see the *v2* image.
+
+[[advanced-deployment-a-b-deployment]]
+== A/B Deployment
+
+A/B deployments generally imply running two (or more) versions of the
+application code or application configuration at the same time for testing or
+experimentation purposes.
+
+The simplest form of an A/B deployment is to divide production traffic between
+two or more distinct *shards* -- a single group of instances with homogeneous
+configuration and code.
+
+More complicated A/B deployments may involve a specialized proxy or load
+balancer that assigns traffic to specific shards based on information about the
+user or application (all "test" users get sent to the B shard, but regular users
+get sent to the A shard).
+
+A/B deployments can be considered similar to A/B testing, although an A/B
+deployment implies multiple versions of code and configuration, where as A/B
+testing often uses one code base with application specific checks.
+
+[[advanced-deployment-when-to-use-a-b-deployment]]
+=== When to Use an A/B Deployment
+
+* When you want to test multiple versions of code or configuration, but are not
+planning to roll one out in preference to the other.
+* When you want to have different configuration in different regions.
+
+An A/B deployment groups different configuration and code -- multiple shards --
+together under a single logical endpoint. Generally, these deployments, if they
+access persistent data, should properly deal with N-1 compatibility (the more
+shards you have, the more possible versions you have running). Use this pattern
+when you need separate internal configuration and code, but end users should not
+be aware of the changes.
+
+[[advanced-deployment-a-b-deployment-example]]
+=== A/B Deployment Example
+
+All A/B deployments are composite deployment types consisting of multiple
+deployment configurations.
+
+[[advanced-deployment-one-service-multiple-deployment-configs]]
+==== One Service, Multiple Deployment Configurations
+
+{product-title}, through labels and deployment configurations, supports multiple
+simultaneous shards being exposed through the same service. To the consuming
+user, the shards are invisible. An example of the simplest possible sharding is
+described below:
+
+. Create the first shard of the application based on the example deployment images:
++
+----
+$ oc new-app openshift/deployment-example --name=ab-example-a --labels=ab-example=true SUBTITLE="shard A"
+----
+
+. Edit the newly created shard to set a label `ab-example=true` that will be
+common to all shards:
++
+----
+$ oc edit dc/ab-example-a
+----
++
+In the editor, add the line `ab-example: "true"` underneath `spec.selector` and
+`spec.template.metadata.labels` alongside the existing
+`deploymentconfig=ab-example-a` label. Save and exit the editor.
+
+. Trigger a re-deployment of the first shard to pick up the new labels:
++
+----
+$ oc deploy ab-example-a --latest
```

Comment:
The label in https://trello.com/c/uQP2AxXy/508-8-deploy-most-recently-available-image is wrong since it was added in 3.4. It would be nice to switch this back to `rollout latest` for 3.4 docs since we want people to start using `rollout` and deprecate `deploy`.